arch/arm/core/irq_init.c

```diff
@@ -33,11 +33,17 @@
  * @return N/A
  */
 
+/*
+ * If Multi-level interrupt controller support is enabled CONFIG_NUM_IRQS may
+ * exceed the maximum number of Cortex-Mx NVIC external inputs.
+ */
 void z_IntLibInit(void)
 {
 	int irq = 0;
 
 	for (; irq < CONFIG_NUM_IRQS; irq++) {
-		NVIC_SetPriority((IRQn_Type)irq, _IRQ_PRIO_OFFSET);
+		if (irq < CM_MAX_NVIC_INPUTS) {
+			NVIC_SetPriority((IRQn_Type)irq, _IRQ_PRIO_OFFSET);
```

Comment:
Is it safe to do this for IRQs which are less than the maximum but not defined by the SoC? I.e. are the writes ignored, and is that guaranteed at architecture level?

Reply:
I believe they are ignored, but still I wouldn't want to do it this way. We should not be writing on non-existing NVIC lines.

Reply:
Instead, I think, we should be keeping (as a DTS or Kconfig) the number of NVIC IRQ lines, as well.